### ISSUE TITLE
Why is the wrong path to the project directory used when downloading?

### ISSUE THREAD

**Issue by ergodeath** [ultralytics/ultralytics#20140]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I first started my project in the folder “C:\HOME\PROG\”.
After that I created a brand new project in the folder “C:\HOME\Projects\Dog Walk”, but the download process takes place at the old address. 
Maybe the problem is in the cache and it needs to be cleaned?

Ive also tried running in a virtual environment, but everything happens the same way.

Python version:
`3.11.4`

Version packages:
```
torch==2.6.0
opencv-python==4.11.0.86
ultralytics==8.3.107
```

Code:
```python
import cv2
import torch
from ultralytics import YOLO

def main():
    model = YOLO("yolo11n-pose.pt")
    device = 'cuda' if torch.cuda.is_available() else 'cpu'
    results = model.train(data="dog-pose.yaml", epochs=3, imgsz=640, device=device) 

    video_path = r"C:\HOME\Projects\Dog Walk\video\slow-motion.mp4"
    cap = cv2.VideoCapture(video_path)
    while cap.isOpened():
        success, frame = cap.read()
        if success:
            results = model.track(frame, persist=True)

            annotated_frame = results[0].plot()

            cv2.imshow("YOLO11 Tracking", annotated_frame)

            if cv2.waitKey(10) & 0xFF == ord("q"):
                break
        else:
            break
    cap.release()
    cv2.destroyAllWindows()

if __name__ == '__main__':
    main()
```

dog-pose.yaml file:

```yaml
# Ultralytics 🚀 AGPL-3.0 License - https://ultralytics.com/license

# Dogs dataset http://vision.stanford.edu/aditya86/ImageNetDogs/ by Stanford
# Documentation: https://docs.ultralytics.com/datasets/pose/dog-pose/
# Example usage: yolo train data=dog-pose.yaml
# parent
# ├── ultralytics
# └── datasets
#     └── dog-pose  ← downloads here (337 MB)

# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: ../datasets/dog-pose # dataset root dir
train: train # train images (relative to 'path') 6773 images
val: val # val images (relative to 'path') 1703 images

# Keypoints
kpt_shape: [24, 3] # number of keypoints, number of dims (2 for x,y or 3 for x,y,visible)

# Classes
names:
  0: dog

# Download script/URL (optional)
download: https://github.com/ultralytics/assets/releases/download/v0.0.0/dog-pose.zip
```

Terminal:
```
PS C:\HOME\Projects\Dog Walk> python -u "c:\HOME\Projects\Dog Walk\main.py"
Ultralytics 8.3.107 🚀 Python-3.11.4 torch-2.6.0+cpu CPU (AMD Ryzen 5 5600H with Radeon Graphics)
engine\trainer: task=pose, mode=train, model=yolo11n-pose.pt, data=dog-pose.yaml, epochs=3, time=None, patience=100, batch=16, 
imgsz=640, save=True, save_period=-1, cache=False, device=cpu, workers=8, project=None, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, 
show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=None, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs\pose\train
Dataset 'dog-pose.yaml' images not found ⚠️, missing path 'C:\HOME\PROG\datasets\dog-pose\val'
Downloading https://ultralytics.com/assets/dog-pose.zip to 'C:\HOME\PROG\datasets\dog-pose.zip'...
100%|...| 337M/337M [00:31<00:00, 11.4MB/s]
Unzipping C:\HOME\PROG\datasets\dog-pose.zip to C:\HOME\PROG\datasets\dog-pose...: 100%|...| 16961/16961 [00:07<00:00, 
Dataset download success ✅ (39.8s), saved to C:\HOME\PROG\datasets
...
...
```

the following folders are created in the project directory `"C:\HOME\Projects\Dog Walk\"`:
```
 runs/pose/train
 runs/pose/train ...
```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ergodeath, thank you for sharing your issue and for your interest in Ultralytics 🚀!

We recommend reviewing the [Ultralytics Docs](https://docs.ultralytics.com/) for more information and helpful examples, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage guides. You may find answers to similar questions there.

From your description, it seems like this could be a 🐛 bug or a configuration issue. If so, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug the problem effectively?

Additionally:
1. Ensure you're using the latest version of the `ultralytics` package by upgrading with:
   ```bash
   pip install -U ultralytics
   ```
2. Confirm your Python environment meets the [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), specifically **Python>=3.8** and **PyTorch>=1.8**.

For custom training-related ❓ questions, please provide more details, including:
- Dataset setup (e.g., paths, structure, or any modifications made)
- Training logs
- Any unique configurations or steps that might help us reproduce the issue

You can also explore various verified environments for running YOLO, such as:
- **Notebooks** with free GPUs: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

For community support or discussions:
- Join our [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time chat
- Visit [Discourse](https://community.ultralytics.com/) for in-depth discussions
- Share your experience or questions on our [Subreddit](https://reddit.com/r/Ultralytics)

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If the CI badge above is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing, ensuring functionality across all YOLO modes and tasks in different environments.

This is an automated response to guide you toward solutions, but rest assured that an Ultralytics engineer will review your issue and assist you further soon. Thank you for your patience and for being part of the Ultralytics community! 🚀✨

**Comment by ergodeath** [comment]
Ive also tried running in a virtual environment, but everything happens the same way

**Comment by ergodeath** [comment]
and the following folders are created in the project directory "C:\HOME\Projects\Dog Walk\":
runs
  - pose
    -train
    -train2

**Comment by Y-T-G** [comment]
You can update the directory using `yolo settings`

https://docs.ultralytics.com/quickstart/#modifying-settings

**Comment by ergodeath** [comment]
Thank you, that helped me!

**Comment by glenn-jocher** [comment]
Great to hear the settings update resolved your issue! For anyone else encountering similar path problems: Ultralytics stores dataset locations in persistent settings that follow you across projects. The `yolo settings datasets_dir='/new/path'` command is perfect for redirecting downloads to your preferred location. You can also check your current paths anytime with just `yolo settings`.

**Comment by OtwakO** [comment]
Is it possible to set these settings like runs_dir in a per project basis in isolation? like a per project overrides without changing the main one in %appdata%/ultralytics/settings.json?

**Comment by glenn-jocher** [comment]
Settings are intentionally global/persistent (written to `settings.json`), so there isn’t a built-in “per project settings profile” that overrides `runs_dir` without touching the main file (see the [SettingsManager reference](https://docs.ultralytics.com/reference/utils/__init__/#class-ultralyticsutilsSettingsManager)), but you can override the output location per run by passing `project=` (and optionally `name=`), which is what `get_save_dir()` uses when creating `runs/...` (see the [get_save_dir reference](https://docs.ultralytics.com/reference/cfg/__init__/#ultralyticscfgget_save_dir)).

```bash
yolo pose train model=yolo26n-pose.pt data=dog-pose.yaml project="C:/HOME/Projects/Dog Walk/runs" name=train
```

```python
from ultralytics import YOLO

model = YOLO("yolo26n-pose.pt")
model.train(data="dog-pose.yaml", project=r"C:\HOME\Projects\Dog Walk\runs", name="train")
```
